wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -2592,6 +2887,215 @@ private function translate_show_like_or_where_condition( WP_Parser_Node $like_or
+			// This is the optional "INSERT INTO ... (field1, field2, ...)" list.
+			foreach ( $fields_node->get_child_nodes() as $field ) {
+				$insert_list[] = $this->unquote_sqlite_identifier( $this->translate( $field ) );
+			}
+		} else {
+			// When no explicit field list is provided, all columns are required.
+			foreach ( array_column( $columns, 'COLUMN_NAME' ) as $column_name ) {
+				$insert_list[] = $column_name;
+			}
+		}
+
+		// 3. Get the list of column names returned by VALUES or SELECT clause.
+		$select_list = array();
+		if ( 'insertQueryExpression' === $node->rule_name ) {
+			// When inserting from a SELECT query, we don't know the column names.
+			// Let's wrap the query with a SELECT (...) LIMIT 0 to get obtain them.
+			$expr = $node->get_first_child_node( 'queryExpressionOrParens' );
+			$stmt = $this->execute_sqlite_query(
+				'SELECT * FROM (' . $this->translate( $expr ) . ') LIMIT 1'
+			);
+			$stmt->execute();
+
+			for ( $i = 0; $i < $stmt->columnCount(); $i++ ) {
+				$select_list[] = $stmt->getColumnMeta( $i )['name'];
+			}
+		} else {
+			// When inserting from a VALUES list, SQLite uses "columnN" naming.
+			foreach ( array_keys( $insert_list ) as $position ) {
+				$select_list[] = 'column' . ( $position + 1 );
+			}
+		}
+
+		// 4. Compose a new INSERT field list with all columns from the table.
+		$fragment = '(';
+		foreach ( $columns as $i => $column ) {
+			$fragment .= $i > 0 ? ', ' : '';
+			$fragment .= $this->quote_sqlite_identifier( $column['COLUMN_NAME'] );
+		}
+		$fragment .= ')';
+
+		// 5. Compose a wrapper SELECT statement emulating IMPLICIT DEFAULT values.
+		$fragment .= ' SELECT ';
+		foreach ( $columns as $i => $column ) {
+			$is_omitted = ! in_array( $column['COLUMN_NAME'], $insert_list, true );
+			$fragment  .= $i > 0 ? ', ' : '';
+			if ( $is_omitted ) {
+				// When a column value is omitted from the INSERT statement, we
+				// need to use the DEFAULT value or the IMPLICIT DEFAULT value.
+				$is_auto_inc = str_contains( $column['EXTRA'], 'auto_increment' );
+				$is_nullable = 'YES' === $column['IS_NULLABLE'];
+				$default     = $column['COLUMN_DEFAULT'];
+				if ( null === $default && ! $is_nullable && ! $is_auto_inc ) {
+					$default = self::DATA_TYPE_IMPLICIT_DEFAULT_MAP[ $column['DATA_TYPE'] ] ?? null;
+				}
+				$fragment .= null === $default ? 'NULL' : $this->pdo->quote( $default );
+			} else {
+				// When a colum value is included, we can use it without change.
+				$position  = array_search( $column['COLUMN_NAME'], $insert_list, true );
+				$fragment .= $this->quote_sqlite_identifier( $select_list[ $position ] );
+			}
+		}
+
+		// 6. Wrap the original insert VALUES or SELECT expression in a FROM clause.
+		$values    = 'insertFromConstructor' === $node->rule_name
+			? $node->get_first_child_node( 'insertValues' )
+			: $node->get_first_child_node( 'queryExpressionOrParens' );
+		$fragment .= ' FROM (' . $this->translate( $values ) . ') WHERE true';
```

Comment:
Why do we need WHERE true? Is this stitched with `AND ...` later on? Also, TIL we can do a subquery like that without giving it an alias.

Reply:
@adamziel It was failing without that, and [then I found it documented](https://www.sqlite.org/lang_insert.html):

<img width="777" alt="Screenshot 2025-03-25 at 15 47 05" src="https://github.com/user-attachments/assets/8736ec78-2ee9-43bb-9237-7ddad5fbed2c" />

Reply:
TIL. Let's leave a comment with a link inline – it's the kind of thing that seems too easy to remove when not careful

Reply:
@adamziel 👍  Added in https://github.com/Automattic/sqlite-database-integration/commit/3faa5533f5ad1b9299430d54a263c3383cce011d.